server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java

```diff
@@ -334,6 +335,46 @@ public boolean isInSortOrderExecutionRequired() {
             return false;
         }
 
+        /**
+         * Return true if any of the builders is a terms aggregation with min_doc_count=0
+         */
+        public boolean hasZeroMinDocTermsAggregation() {
+            final Queue<AggregationBuilder> queue = new LinkedList<>(aggregationBuilders);
+            while (queue.isEmpty() == false) {
+                final AggregationBuilder current = queue.poll();
+                if (current == null) {
+                    continue;
+                }
+                if (current instanceof TermsAggregationBuilder termsBuilder) {
+                    if (termsBuilder.minDocCount() == 0) {
+                        return true;
+                    }
+                }
+                queue.addAll(current.getSubAggregations());
+            }
+            return false;
+        }
```

Comment:
I wouldn't worry too much about the recursion. The tree isn't that deep and other things recur on the tree. Like running aggs.

Reply:
I would prefer to leave this iterative unless there are strong objections.

Reply:
No objections